application/controllers/admin/database.php

```diff
@@ -814,6 +814,8 @@ private function actionUpdateSurveyLocaleSettings($iSurveyID)
         $languagelist = $oSurvey->additionalLanguages;
         $languagelist[] = $oSurvey->language;
 
+        $saveSurvey = true;
```

Comment:
Please do not introduce needless state into the logic. This is spaghetti code. Instead, throw an exception if survey can't be saved.